api/src/main/java/org/apache/iceberg/expressions/InclusiveMetricsEvaluator.java

```diff
@@ -254,5 +258,43 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
     public <T> Boolean notIn(BoundReference<T> ref, Literal<T> lit) {
       return ROWS_MIGHT_MATCH;
     }
+
+    @Override
+    public <T> Boolean startsWith(BoundReference<T> ref, Literal<T> lit) {
+      int id = ref.fieldId();
+
+      String prefix = (String) lit.value();
+      ByteBuffer prefixAsBytes = ByteBuffer.wrap(prefix.getBytes(StandardCharsets.UTF_8));
+
+      Comparator<ByteBuffer> comparator = Comparators.unsignedBytes();
+
+      if (lowerBounds != null && lowerBounds.containsKey(id)) {
+        ByteBuffer lower = lowerBounds.get(id);
+        // take the same number of bytes from prefix and lower bound
+        int length = Math.min(prefixAsBytes.remaining(), lower.remaining());
+
+        int cmp = comparator.compare(
+            length < lower.remaining() ? BinaryUtil.truncateBinary(lower, length) : lower,
```

Comment:
It isn't safe to call `truncateBinary` on a string value because it truncates by bytes. Strings are truncated by unicode codepoints instead.

Reply:
Also, the `truncate` methods will return the original value if there is no need to truncate, so you can simplify this code.

Reply:
`BinaryUtil.truncateBinary` will complain if the length is lower than the number of remaining elements. I would probably change that to return the original value.

Reply:
Yeah, I think so.